src/pages/index.js

```diff
@@ -73,29 +155,75 @@ const newCardPopup = new PopupWithForm(
 newCardPopup.setEventListeners();
 
 function handleAddNewCardSubmit(data) {
-  const card = createCard({ name: data.name, link: data.url });
-  cardSection.addItem(card);
+  // const card = createCard({ name: data.name, link: data.url });
+  newCardPopup.submitText("Saving...");
+  api
+    .createCard(data)
+    .then((data) => createCard(data))
+    .then((data) => cardSection.addItem(data))
+    .then(() => newCardPopup.close())
+    .catch((err) => {
+      console.error(err);
+    })
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors